src/main/java/org/microg/nlp/backend/nominatim/BackendService.java

```diff
@@ -182,7 +210,7 @@ public void run() {
             synchronized (done) {
                 try {
                     Log.d(TAG, "Requesting " + url);
-                    HttpURLConnection connection = (HttpURLConnection) new URL(url)
+                    HttpsURLConnection connection = (HttpsURLConnection) new URL(url)
```

Comment:
You can keep this at HttpURLConnection as you don't use any feature of HttpsURLConnection. Casting this to a HttpsURLConnection instance does not improve security ;)